api/v1/common/sagemaker_api.go

```diff
@@ -64,6 +72,68 @@ type Channel struct {
 	ShuffleConfig *ShuffleConfig `json:"shuffleConfig,omitempty"`
 }
 
+type ProcessingInput struct {
+	InputName string            `json:"inputName"`
+	S3Input   ProcessingS3Input `json:"s3Input"`
+}
+
+type ProcessingS3Input struct {
+	LocalPath LocalPath `json:"localPath"`
+
+	CompressionType CompressionType `json:"compressionType,omitempty"`
+
+	// +kubebuilder:validation:Enum=FullyReplicated;ShardedByS3Key
+	S3DataDistributionType S3DataDistributionType `json:"s3DataDistributionType,omitempty"`
+
+	// +kubebuilder:validation:Enum=S3Prefix;ManifestFile
+	S3DataType string `json:"s3DataType"`
+
+	// +kubebuilder:validation:Enum=Pipe;File
+	S3InputMode TrainingInputMode `json:"s3InputMode"`
+
+	S3Uri S3Uri `json:"s3Uri"`
+}
+
+type ProcessingOutputConfig struct {
+	// +kubebuilder:validation:MaxLength=1024
+	KmsKeyId string `json:"kmsKeyId,omitempty"`
+
+	// +kubebuilder:validation:MaxItems=10
+	Outputs []ProcessingOutputStruct `json:"outputs"`
+}
+
+type ProcessingOutputStruct struct {
+	OutputName string `json:"outputName"`
+
+	S3Output ProcessingS3Output `json:"s3Output"`
+}
+
+type ProcessingS3Output struct {
+	LocalPath LocalPath `json:"localPath"`
+
+	// +kubebuilder:validation:Enum=Continuous;EndOfJob
+	S3UploadMode string `json:"s3UploadMode"`
+
+	S3Uri S3Uri `json:"s3Uri"`
+}
+
+type ProcessingNetworkConfig struct {
```

Comment:
It's so odd that this is a sub-field of Processing whereas its in the root of the request for all other types.

Reply:
It is but this one is cleaner as well